src/app/core/services/auth.service.spec.ts

```diff
@@ -18,16 +20,95 @@ describe('AuthService', () => {
-      imports: [HttpClientTestingModule, NgxPermissionsModule.forRoot(), RouterTestingModule.withRoutes(routes)],
-      providers: [AuthService, NgxPermissionsService],
+      imports: [
+        HttpClientTestingModule,
+        NgxPermissionsModule.forRoot(),
+        RouterTestingModule.withRoutes(routes)
+      ],
+      providers: [
+        AuthService, NgxPermissionsService,
+        {provide: HttpClient, useValue: httpSpy},
+        {provide: Router, useValue: routerServiceSpy}
+      ],
       declarations: [FakeComponent]
     });
     service = TestBed.inject(AuthService);
   });
 
+  afterEach(() => {
+    localStorage.clear();
+    service.logout();
+    httpSpy.post.and.returnValue(of(adminTokenResponse));
+    routerServiceSpy.navigateByUrl.calls.reset();
+  });
+
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should set guest permissions if token is absent', async() => {
+    expect(service.permissions).toEqual([UserTypesEnum.GUEST]);
+  });
+
+  it('should set encoded token in localstorage', async () => {
+    service.login({}).subscribe((token) => {
+      expect(localStorage.getItem('token')).toEqual(token.Token);
+    });
+  });
+
+  it('should set admin permissions if token with admin utp was received', async () => {
+    service.login({}).subscribe(() => {
+      expect(service.permissions).toEqual([UserTypesEnum.ADMIN]);
+    });
+  });
+
+  it('should set regular user permissions if token with regular user permissions was received', async () => {
+    const regularTokenResponse = {Token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MDUxNzk2OTYsImlhdCI6MTYwNTE3ODc5NiwibmJmIjoxNjA1MTc4Nzk2LCJzdWIiOiJhZG1pbiIsInVpZCI6MSwidXRwIjpudWxsLCJwZXJtIjpbInJlZ3VsYXIiXX0.p5uzf6PUVGup4IbW8eSo3osAauvWK1nogzxfxwPmvF0'};
+    httpSpy.post.and.returnValue(of(regularTokenResponse));
+
+    service.login({}).subscribe(() => {
+      expect(service.permissions).toEqual([UserTypesEnum.REGULAR]);
+    });
+  });
+
+  it('should set auditor user permissions if token with auditor user permissions was received', async () => {
+    const regularTokenResponse = {Token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MDUxNzk2OTYsImlhdCI6MTYwNTE3ODc5NiwibmJmIjoxNjA1MTc4Nzk2LCJzdWIiOiJhZG1pbiIsInVpZCI6MSwidXRwIjpudWxsLCJwZXJtIjpbImF1ZGl0b3IiXX0.e8_h-HI1x0dDO6kwdbFA_zk11MCPP_AFOhPyc7df_bM'};
+    httpSpy.post.and.returnValue(of(regularTokenResponse));
+
+    service.login({}).subscribe(() => {
+      expect(service.permissions).toEqual([UserTypesEnum.AUDITOR]);
+    });
+  });
+
+  it('should redirect to jobs', async () => {
+    service.login({}).subscribe(() => {
+      expect(routerServiceSpy.navigateByUrl).toHaveBeenCalledWith('/jobs');
+    });
+  });
+
+  it('should clear token from localStorage upon logout', async () => {
+    service.logout();
+
+    expect(localStorage.getItem('token')).toBeFalsy();
+  });
+
+  it('should set guest user permission after logout', async () => {
```

Comment:
Note to self: in case of this test it would be good to first log in, then log out, and check the permissions after that. Logout is executed after every test to additional call in this spec doesn't change anything.